airflow/gcp/operators/dataproc.py

```diff
@@ -576,126 +516,136 @@ class DataprocClusterScaleOperator(DataprocOperationBaseOperator):
     @apply_defaults
     def __init__(self,
                  cluster_name: str,
-                 project_id: str,
+                 project_id: Optional[str] = None,
                  region: str = 'global',
                  num_workers: int = 2,
                  num_preemptible_workers: int = 0,
-                 graceful_decommission_timeout: Optional[str] = None,
+                 gcp_conn_id: str = "google_cloud_default",
                  *args,
                  **kwargs) -> None:
-        super().__init__(project_id=project_id, region=region, *args, **kwargs)
+        super().__init__(*args, **kwargs)
+        self.project_id = project_id
+        self.region = region
         self.cluster_name = cluster_name
         self.num_workers = num_workers
         self.num_preemptible_workers = num_preemptible_workers
-
-        # Optional
-        self.optional_arguments = {}  # type: Dict
-        if graceful_decommission_timeout:
-            self.optional_arguments['gracefulDecommissionTimeout'] = \
-                self._get_graceful_decommission_timeout(
-                    graceful_decommission_timeout)
+        self.gcp_conn_id = gcp_conn_id
 
     def _build_scale_cluster_data(self):
         scale_data = {
             'config': {
-                'workerConfig': {
-                    'numInstances': self.num_workers
+                'worker_config': {
+                    'num_instances': self.num_workers
                 },
-                'secondaryWorkerConfig': {
-                    'numInstances': self.num_preemptible_workers
+                'secondary_worker_config': {
+                    'num_instances': self.num_preemptible_workers
                 }
             }
         }
         return scale_data
 
-    @staticmethod
-    def _get_graceful_decommission_timeout(timeout):
```

Comment:
This and optional_arguments were never used.

Reply:
Good point! I missed the fact that those params were passed using `**self.optional_arguments` ✅